lib/Bio/EnsEMBL/DataCheck/Checks/CheckTableSizes.pm

```diff
@@ -0,0 +1,95 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckTableSizes;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::DataCheck::Utils qw/ array_diff /;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckTableSizes',
+  DESCRIPTION    => 'Tables must be populated and not differ significantly in row numbers',
+  GROUPS         => ['compara', 'compara_tables'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+};
+
+sub tests {
+  my ($self) = @_;
+  
+  my $curr_dba = $self->dba;
+  my $curr_helper = $curr_dba->dbc->sql_helper;
+  my $prev_dba = $self->registry->get_DBAdaptor('compara_prev', 'compara') || $self->get_old_dba;
+  my $prev_helper = $prev_dba->dbc->sql_helper;
+  
+  my $table_sql = "SHOW TABLES";
+  my $curr_tables = $curr_helper->execute_simple( -SQL => $table_sql );
+  my $prev_tables = $prev_helper->execute_simple( -SQL => $table_sql );
+  my $curr_db_name = $curr_dba->dbc->dbname;
+  my $prev_db_name = $prev_dba->dbc->dbname;
+  
+  my %prev_tables = ();
+  foreach my $table ( @$prev_tables ) {
+    $prev_tables{$table}++;
+  }
+  my %curr_tables = ();
+  foreach my $table ( @$curr_tables ) {
+    $curr_tables{$table}++;
+  }
+  
+  foreach my $table ( @$curr_tables ) {
+    my $desc_1 = "The number of rows in $table for $curr_db_name has not increased by >10% from $prev_db_name";
+    my $desc_2 = "The number of rows in $table for $curr_db_name has not decreased by >5% from $prev_db_name";
+    my $desc_3 = "The number of rows in $table has not remained static between $curr_db_name and $prev_db_name";
+    my $sql = qq/
+      SELECT COUNT(*) FROM $table
+    /;
+      
+    if ( exists($prev_tables{$table}) ) {
+      my $prev_row_count = $prev_helper->execute_single_result( -SQL => $sql );
+      my $curr_row_count = $curr_helper->execute_single_result( -SQL => $sql );
+
+      cmp_ok( $curr_row_count, '<=', ($prev_row_count*1.1), $desc_1 );
+      cmp_ok( $curr_row_count, '>=', ($prev_row_count*0.95), $desc_2 );
+      cmp_ok( $curr_row_count, '!=', $prev_row_count, $desc_3 );
+        
+    } else {
+      my $desc_4 = "Table: $table is new in $curr_db_name";
+      pass( $desc_4 );
+      my $desc_5 = "New table: $table is populated with data";
+      is_rows_nonzero($curr_dba->dbc, $sql, $desc_5);
+    }
+  }
+  
+  foreach my $table ( @$prev_tables ) {
+    my $desc_5 = "Table $table is missing in $curr_db_name and present in $prev_db_name";
+    unless ( exists($curr_tables{$table}) ) {
+      fail($desc_5);
```

Comment:
Generally speaking, it's best to avoid explicit calls to `fail` (and `pass`) (cf Perl's Test::More docs, at the bottom of this section: https://perldoc.perl.org/Test/More.html#I'm-ok%2c-you're-not-ok.)
Here, you could just have the `exists($curr_tables{$table})` in an `ok` statement, and remove the conditional loop and the need for `fail`.
(Up above, it seems like the `pass` isn't really necessary, the test on line 81 implicitly tells you that. It's also a bit confusing to use `$desc_5` for two different descriptions.)

Reply:
I will fix some of these stylistic changes before you merge, I prefer your idea here